plugins/inputs/system/ps.go

```diff
@@ -64,6 +64,27 @@ func (s *SystemPS) CPUTimes(perCPU, totalCPU bool) ([]cpu.TimesStat, error) {
 	return cpuTimes, nil
 }
 
+type set struct {
+	m map[string]struct{}
+}
+
+func (s *set) add(key string) {
+	s.m[key] = struct{}{}
+}
+
+func (s *set) has(key string) bool {
+	var ok bool
+	_, ok = s.m[key]
+	return ok
+}
+
+func newSet() *set {
+	s := &set{
+		m: make(map[string]struct{}),
+	}
+	return s
+}
+
```

Comment:
Can we have this as an internal type as we often see `set` operations... We can add `intersection` and `union` later on-demand as I think we have it in the code in multiple locations...

Reply:
I don't mind if we move it to internal and extend it. Next time you (or anyone) needs it, feel free to move it around.